root/root.go

```diff
@@ -149,7 +192,8 @@ func Run(configFileName string) (*common.RestServiceInfo, error) {
 	rootService := &Root{}
 	rootServiceConfig := common.ServiceConfig{
 		Common:          fullConfig.Services["root"].Common,
-		ServiceSpecific: make(map[string]interface{})}
+		ServiceSpecific: make(map[string]interface{})
```

Comment:
Oh this goes way back... So all services in their config have a section that every service has - e.g., what host and port to listen on, etc. Then optionally they can have a section that is only specific to them. (In case of root, it's convoluted - the entire config is specific to it). It's this convoluted way of keeping things both easy to read in yaml, but easy to use in code because instead of maps the are actual structures with fields that can be checked statically and all that. At the expense of slightly convoluted conversion from one to another.